### PR TITLE
feedback notes, use what feels good, ignore what doesn't :)

### PR DESCRIPTION
**Pull request by mewmew**

### CODE REVIEW

File: shorter.go [hunk 1]
```diff
@@ -320,7 +337,8 @@ func handleRequests(w http.ResponseWriter, r *http.Request, indexTmpl *template.
 		switch requestType {
 		case "url":
 			formURL := r.Form.Get("url")
-			if len(formURL) < 10 || formURL[:7] != "http://" && formURL[:8] != "https://" {
+			// TODO: where does the 10 come from in len(formURL) < 10? consider using e.g. len("ftp://x.se") or something instead of magic number. Perhaps document why we do this check? Is it to prevent recursion? Is to to only handle valid links to other pages?
```
Comment (7i) [inline shorter.go]:
```suggestion
			// simple sanity check to fail early, If len(formURL) is less than 11 it is definitely an invalid url.
```

File: shorter.go [hunk 2]
```diff
@@ -320,7 +337,8 @@ func handleRequests(w http.ResponseWriter, r *http.Request, indexTmpl *template.
...
+			if len(formURL) < 10 || !strings.HasPrefix(formURL, "http://") && !strings.HasPrefix(formURL, "https://") {
```
Comment (7i) [inline shorter.go]:
```suggestion
			if len(formURL) < 11 || !strings.HasPrefix(formURL, "http://") && !strings.HasPrefix(formURL, "https://") {
```

File: shorter.go [hunk 3]
```diff
@@ -284,6 +300,7 @@ func handleRequests(w http.ResponseWriter, r *http.Request, indexTmpl *template.
 	if r.Method == http.MethodPost {
 		err := r.ParseMultipartForm(config.MaxFileSize)
 		if err != nil {
+			// TODO: consider logging error to disk, and only output static error message to use. This is to prevent information disclosure.
```
Comment (7i) [inline shorter.go]:
```suggestion
			// TODO: all logs should be logged to disk
```

File: shorter.go [hunk 4]
```diff
@@ -332,9 +350,10 @@ func handleRequests(w http.ResponseWriter, r *http.Request, indexTmpl *template.
 			newLink := &link{linkType: "url", data: formURL, times: xTimes, timeout: time.Now().Add(currentLinkLen.timeout)}
 			key, err := currentLinkLen.Add(newLink)
 			if err != nil {
+				// NOTE: as all errors of Add are controlled by us, this is not an information disclosure. However, it could become if e.g. file handling was added to the Add method, and we return an error from the os. To limit this risk, we could define specific errors that are intended for the user, e.g. ErrNoKeysLeft, ErrTimeout, etc. Each of these would have static error messages. Then for any other error we just log to disk, and return internal server error to the user.
```
Comment (7i) [inline shorter.go]:
```suggestion
				// TODO: log all errors to disk and only return generic static errors to users e.g. ErrNoKeysLeft, ErrTimeout, etc.
```

File: shorter.go [hunk 5]
```diff
@@ -284,6 +300,7 @@ func handleRequests(w http.ResponseWriter, r *http.Request, indexTmpl *template.
 	if r.Method == http.MethodPost {
 		err := r.ParseMultipartForm(config.MaxFileSize)
 		if err != nil {
+			// TODO: all logs should be logged to disk
```
Comment (mewmew) [inline shorter.go:303]:
```suggestion
			// TODO: all errors should be logged to disk
```

**Review by 7i — APPROVED** [review]

**Comment by mewmew** [comment]
:tada:

File: shorter.go [hunk 6]
```diff
@@ -20,20 +20,26 @@ import (
...
+	mutationsLen1 = len(charset)
+	mutationsLen2 = len(charset) * len(charset)
+	mutationsLen3 = len(charset) * len(charset) * len(charset)
+)
+
 const debug = true
 
-// Config contains all valid fealds from a shorter config file
+// dateFormat specifies the format in which date and time is represented.
+const dateFormat = "Mon 2006-01-02 15:04 MST"
```
Comment (mewmew) [inline shorter.go:34]: I looked to see if this was a standard format, but seems like you've created a new standard to unify them all :)

From `go doc time.ANSIC`:

```go
const (
	ANSIC       = "Mon Jan _2 15:04:05 2006"
	UnixDate    = "Mon Jan _2 15:04:05 MST 2006"
	RubyDate    = "Mon Jan 02 15:04:05 -0700 2006"
	RFC822      = "02 Jan 06 15:04 MST"
	RFC822Z     = "02 Jan 06 15:04 -0700" // RFC822 with numeric zone
	RFC850      = "Monday, 02-Jan-06 15:04:05 MST"
	RFC1123     = "Mon, 02 Jan 2006 15:04:05 MST"
	RFC1123Z    = "Mon, 02 Jan 2006 15:04:05 -0700" // RFC1123 with numeric zone
	RFC3339     = "2006-01-02T15:04:05Z07:00"
	RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
)
```

**Pull request merged**